api/util.js

```diff
@@ -38,5 +38,22 @@ function assign(target, src) {
   return target;
 }
 
+/**
+ * Get the current location. More readily mocked than using the global
+ * directly.
+ *
+ * @return {Location} The current location.
+ */
+function currentLocation() {
+  /* location is tricky to mock in the browser */
+  /* istanbul ignore if */
```

Comment:
This module could export `setLocation`, `getLocation`, and `clearLocation` (`setLocation` and `clearLocation` would be used in tests).